src/app/shared/components/course-form/course-form.component.html

```diff
@@ -1,60 +1,127 @@
+          >
+          <span *ngIf="description?.errors?.['required']" class="text-danger"
+            >Description is required.</span
+          >
+        </div>
       </div>
-      <div class="separator"></div>
-      <h3>Authors</h3>
+
+      <h3>Duration</h3>
       <div class="form__control">
         <div class="course__flex">
-          <!--Use the name `author` for the form control.-->
-          <input type="text" id="author" placeholder="Input Author Name" />
-          <app-button id="createAuthor"></app-button>
+          <input
+            type="number"
+            id="duration"
+            placeholder="Input duration"
+            formControlName="duration" />
+          <div class="course__duration">
+            {{ courseForm.get('duration')?.value | duration }}
+          </div>
         </div>
-        <span id="authorErrorMessage"
-          >New author should contain only latin letters and numbers.</span
-        >
-        <div class="course__authors">
-          <!--Use the name `authors` for the form array.-->
-          ::Authors::
+        <div
+          id="durationErrorMessage"
+          *ngIf="duration?.invalid && duration?.touched">
+          <span *ngIf="duration?.errors?.['required']" class="text-danger"
+            >Duration is required.</span
+          >
+          <span *ngIf="duration?.errors?.['min']" class="text-danger"
+            >Duration should not be less than 0.</span
+          >
         </div>
       </div>
+
       <div class="separator"></div>
-      <h3>Duration</h3>
-      <div class="form__control">
+      <h3>Authors</h3>
+      <div class="form__control course__panel" formGroupName="newAuthor">
         <div class="course__flex">
-          <!--Use the name `duration` for the form array.-->
-          <input type="number" id="duration" placeholder="Input duration" />
-          <div class="course__duration">::Duration::</div>
+          <input
+            type="text"
+            id="author"
+            placeholder="Input Author Name"
+            formControlName="author" />
+          <app-button
+            id="createAuthor"
+            [buttonText]="ButtonText.createAuthor"
+            (click)="createNewAuthor()"></app-button>
         </div>
-        <span id="durationErrorMessage" class="text-danger"
-          >Duration is required.</span
-        >
+        <div
+          id="authorErrorMessage"
+          *ngIf="newAuthor?.invalid && newAuthor?.touched">
+          <span *ngIf="newAuthor?.errors?.['pattern']" class="text-danger"
+            >New author should contain only latin letters and numbers.</span
+          >
+          <span *ngIf="newAuthor?.errors?.['minlength']" class="text-danger"
+            >New author should not be less than 2 characters.</span
+          >
+        </div>
+
+        <div class="course__authors">
+          <h4>Course Authors</h4>
+          <p *ngIf="courseAuthors.length <= 0">Author list is empty</p>
```

Comment:
Hmmm, I dont think I ever saw an array with negative length. I know I am being peckish on this... 😆 
length = 0 is far enough.